// mmn/atoms/Checkbox.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var _reduxForm = require("redux-form");

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _reactRedux = require("react-redux");

var _uuid = _interopRequireDefault(require("uuid"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  width: calc(100% - 35px);\n  display: inline-block;\n  font-size: 12px;\n  margin-left: 7px;\n  user-select: none;\n  cursor: pointer;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: 2px;\n  left: 2px;\n  width: 10px;\n  height: 10px;\n  background-color: ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  position: relative;\n  width: 14px;\n  height: 14px;\n  border: 1px solid ", ";\n  cursor: pointer;\n  display: inline-block;\n  vertical-align: top;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  line-height: 18px;\n  width: 100%;\n  ", "\n\n  input {\n    display: none;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var Container = _styledComponents.default.div(_templateObject(), function (props) {
  return props.style && (0, _styledComponents.css)(_templateObject2(), props.style);
});

var Box = _styledComponents.default.div(_templateObject3(), function (props) {
  return props.theme.colors.gray4;
});

var Checked = _styledComponents.default.div(_templateObject4(), function (props) {
  return props.theme.colors.red2;
});

var Label = _styledComponents.default.div(_templateObject5());

var Checkbox = function Checkbox(_ref) {
  var _ref$input = _ref.input,
      value = _ref$input.value,
      onChange = _ref$input.onChange,
      name = _ref$input.name,
      input = _objectWithoutPropertiesLoose(_ref$input, ["value", "onChange", "name"]),
      id = _ref.id,
      checked = _ref.checked,
      style = _ref.style,
      formName = _ref.formName,
      disabled = _ref.disabled,
      label = _ref.label,
      changeAction = _ref.changeAction;

  return _react.default.createElement(Container, {
    id: id,
    style: style,
    onClick: function onClick() {
      if (!disabled) changeAction(name, !checked, formName);
    }
  }, _react.default.createElement("input", {
    type: "checkbox",
    name: name,
    onChange: onChange
  }), _react.default.createElement(Box, null, checked && _react.default.createElement(Checked, null)), label && _react.default.createElement(Label, null, label));
};

Checkbox.propTypes = {
  input: _propTypes.PropTypes.shape(_propTypes.PropTypes.any).isRequired,
  id: _propTypes.PropTypes.string,
  checked: _propTypes.PropTypes.bool.isRequired,
  disabled: _propTypes.PropTypes.bool,
  style: _propTypes.PropTypes.shape(_propTypes.PropTypes.any),
  formName: _propTypes.PropTypes.string.isRequired,
  label: _propTypes.PropTypes.string,
  changeAction: _propTypes.PropTypes.func.isRequired
};
Checkbox.defaultProps = {
  id: (0, _uuid.default)(),
  disabled: false,
  label: null,
  style: null
};

var mstp = function mstp(state, ownProps) {
  return {
    checked: (0, _reduxForm.formValueSelector)(ownProps.formName)(state, ownProps.name)
  };
};

var mdtp = function mdtp(dispatch) {
  return {
    changeAction: function changeAction(field, value, formName) {
      return dispatch((0, _reduxForm.change)(formName, field, value));
    }
  };
};

var _default = (0, _reactRedux.connect)(mstp, mdtp)(function (props) {
  return _react.default.createElement(_reduxForm.Field, _extends({}, props, {
    component: Checkbox
  }));
});

exports.default = _default;
