// mmn/atoms/Input.stories.js
"use strict";

exports.__esModule = true;
exports.validateEmail = void 0;

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@storybook/react");

var _addonKnobs = require("@storybook/addon-knobs");

var _validator = require("validator");

var _customDecorators = require("../../.storybook/customDecorators");

var _theme = _interopRequireDefault(require("../theme"));

var _Input = _interopRequireDefault(require("./input/Input"));

var _InputHoverPersist = _interopRequireDefault(require("./input/InputHoverPersist"));

var _InputAgeGate = _interopRequireDefault(require("./input/InputAgeGate"));

var _InputFooterSubscription = _interopRequireDefault(require("./input/InputFooterSubscription"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable no-alert */
var LockSrc = function LockSrc(props) {
  return _react.default.createElement("svg", props, _react.default.createElement("path", {
    d: "M12 5.5a4.5 4.5 0 0 0-9 0s9-.113 9 0z",
    stroke: "#7C7C7C"
  }), _react.default.createElement("rect", {
    x: ".5",
    y: "5.5",
    width: "14",
    height: "11",
    rx: "2.5",
    stroke: "#7C7C7C"
  }), _react.default.createElement("rect", {
    x: "7",
    y: "10",
    width: "1",
    height: "4",
    rx: ".5",
    fill: "#7C7C7C"
  }), _react.default.createElement("path", {
    d: "M8.5 9.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0z",
    stroke: "#7C7C7C"
  }));
};

LockSrc.defaultProps = {
  width: "15",
  height: "17",
  viewBox: "0 0 15 17",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
};

var MailSrc = function MailSrc(props) {
  return _react.default.createElement("svg", props, _react.default.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M1.188.792a.397.397 0 0 0-.396.395V11.48c0 .218.178.396.395.396h16.625a.397.397 0 0 0 .396-.396V1.188a.397.397 0 0 0-.395-.396H1.188zm16.625 11.875H1.188A1.189 1.189 0 0 1 0 11.479V1.188C0 .532.533 0 1.188 0h16.625C18.466 0 19 .533 19 1.188v10.291c0 .655-.533 1.188-1.188 1.188z",
    fill: "#7C7C7C"
  }), _react.default.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M9.5 7.126c-.409 0-.816-.127-1.165-.38L.365.983A.395.395 0 0 1 .278.43.397.397 0 0 1 .83.342L8.8 6.105c.419.305.981.304 1.399 0L18.173.343a.396.396 0 0 1 .464.64l-7.973 5.765c-.349.252-.756.38-1.164.38z",
    fill: "#7C7C7C"
  }), _react.default.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M.608 12.414a.395.395 0 0 1-.233-.716l7.833-5.685a.396.396 0 0 1 .465.64L.84 12.338a.393.393 0 0 1-.232.076zM18.39 12.414a.39.39 0 0 1-.232-.076l-7.831-5.685a.397.397 0 0 1 .466-.64l7.83 5.686a.395.395 0 0 1-.232.715z",
    fill: "#7C7C7C"
  }));
};

MailSrc.defaultProps = {
  width: "19",
  height: "13",
  viewBox: "0 0 19 13",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
};

var GeoSrc = function GeoSrc(props) {
  return _react.default.createElement("svg", props, _react.default.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M10 6.956A3.047 3.047 0 0 0 6.956 10 3.047 3.047 0 0 0 10 13.043 3.047 3.047 0 0 0 13.043 10 3.047 3.047 0 0 0 10 6.956zm0 6.957A3.918 3.918 0 0 1 6.087 10 3.918 3.918 0 0 1 10 6.087 3.918 3.918 0 0 1 13.913 10 3.918 3.918 0 0 1 10 13.913z",
    fill: "#000"
  }), _react.default.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M10 2.609A7.398 7.398 0 0 0 2.609 10a7.398 7.398 0 0 0 7.39 7.391A7.398 7.398 0 0 0 17.392 10 7.398 7.398 0 0 0 10 2.609zm0 15.652c-4.555 0-8.261-3.704-8.261-8.261 0-4.557 3.706-8.26 8.26-8.26 4.556 0 8.262 3.703 8.262 8.26 0 4.556-3.706 8.26-8.261 8.26z",
    fill: "#222"
  }), _react.default.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M10 2.609a.433.433 0 0 1-.435-.435V.434C9.565.192 9.76 0 10 0s.435.191.435.435v1.739a.433.433 0 0 1-.435.435zM19.565 10.435h-1.739a.433.433 0 0 1-.435-.435c0-.244.195-.435.435-.435h1.74c.24 0 .434.191.434.435a.433.433 0 0 1-.435.435zM2.174 10.435H.434A.433.433 0 0 1 0 10c0-.244.195-.435.435-.435h1.739c.24 0 .435.191.435.435a.433.433 0 0 1-.435.435zM10 20a.433.433 0 0 1-.435-.435v-1.739c0-.243.195-.435.435-.435s.435.192.435.435v1.74A.433.433 0 0 1 10 20z",
    fill: "#222"
  }));
};

GeoSrc.defaultProps = {
  width: "20",
  height: "20",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
};
var label = 'Icon';
var options = {
  None: null,
  Lock: LockSrc,
  Mail: MailSrc
}; // eslint-disable-next-line import/prefer-default-export

var validateEmail = function validateEmail(val) {
  return val && !(0, _validator.isEmail)(val) ? 'Please enter a valid email address.' : undefined;
};

exports.validateEmail = validateEmail;
var groupId = 'ALL';

var onClick = function onClick() {
  return alert('clicked');
};

(0, _react2.storiesOf)('form/Input', module).addDecorator(_customDecorators.limitWidth).addWithJSX('Input - Default', function () {
  return _react.default.createElement(_Input.default, {
    name: "Input - Default",
    disabled: (0, _addonKnobs.boolean)('Disabled', false),
    placeholder: (0, _addonKnobs.text)('Placeholder', 'Placeholder'),
    icon: (0, _addonKnobs.select)(label, options, null, groupId),
    backgroundColor: (0, _addonKnobs.text)('Background Color', null, 'COLORS'),
    asteriskText: (0, _addonKnobs.text)('Asterisk', null),
    outlineColor: (0, _addonKnobs.text)('Outline Color', null, 'COLORS'),
    validate: function validate(value) {
      return !value && 'required';
    } // errorMessage={text('Error', 'This is an error message!')}

  });
}).addWithJSX('Input - With Icon and Action', function () {
  return _react.default.createElement(_Input.default, {
    placeholder: (0, _addonKnobs.text)('Placeholder', 'Password'),
    type: "password",
    icon: (0, _addonKnobs.select)(label, options, LockSrc, groupId),
    name: "input",
    action: {
      label: (0, _addonKnobs.text)('Action label', 'Show'),
      onClick: function onClick() {
        return alert("Change type='password' to type='text' here to show the password");
      }
    }
  });
}).addWithJSX('Input - With Email Icon', function () {
  return _react.default.createElement(_Input.default, {
    placeholder: (0, _addonKnobs.text)('Placeholder', 'Email Address'),
    type: "Email",
    icon: (0, _addonKnobs.select)(label, options, MailSrc, groupId),
    name: "input"
  });
}).addWithJSX('Input - With Email Icon and Action', function () {
  return _react.default.createElement(_Input.default, {
    placeholder: (0, _addonKnobs.text)('Placeholder', 'Email Address'),
    type: "input",
    icon: (0, _addonKnobs.select)(label, options, MailSrc, groupId),
    name: "input",
    action: {
      label: (0, _addonKnobs.text)('Action label', 'Re-Send Verification'),
      color: _theme.default.colors.red2,
      onClick: function onClick() {
        return alert("Change type='password' to type='text' here to show the password");
      }
    }
  });
}).addWithJSX('Input - GEO', function () {
  return _react.default.createElement(_Input.default, {
    placeholder: (0, _addonKnobs.text)('Placeholder', 'Location'),
    name: "input",
    action: {
      icon: GeoSrc,
      onClick: function onClick() {
        return alert("Do the GEO thing here");
      }
    }
  });
}).addWithJSX('Input - Button', function () {
  return _react.default.createElement(_Input.default, {
    placeholder: (0, _addonKnobs.text)('Placeholder', 'Input with Button'),
    name: "input",
    button: {
      disabled: (0, _addonKnobs.boolean)('Disable Button', false, 'BUTTON'),
      label: (0, _addonKnobs.text)('Button Label', 'Submit', 'BUTTON'),
      onClick: onClick()
    }
  });
}).addWithJSX('Input Sign In', function () {
  return _react.default.createElement(_InputHoverPersist.default, {
    placeholder: (0, _addonKnobs.text)('Placeholder', 'First Name'),
    type: "input",
    name: "input"
  });
}).addWithJSX('Input Sign In - With Email Icon', function () {
  return _react.default.createElement(_InputHoverPersist.default, {
    placeholder: (0, _addonKnobs.text)('Placeholder', 'Email Address'),
    type: "input",
    icon: (0, _addonKnobs.select)(label, options, MailSrc, groupId),
    name: "input"
  });
}).addWithJSX('Age Gate / Exit Popup', function () {
  return _react.default.createElement(_InputAgeGate.default, {
    name: "input",
    disabled: (0, _addonKnobs.boolean)('Disabled', false),
    placeholder: (0, _addonKnobs.text)('Placeholder', 'Email Address (Optional)'),
    button: {
      label: 'Enter',
      backgroundColor: _theme.default.colors.red2,
      onClick: onClick
    }
  });
}).addWithJSX('Footer Subscription', function () {
  return _react.default.createElement(_InputFooterSubscription.default, {
    name: "input",
    disabled: (0, _addonKnobs.boolean)('Disabled', false),
    placeholder: (0, _addonKnobs.text)('Placeholder', 'Email Address (Optional)'),
    button: {
      label: 'Enter',
      backgroundColor: _theme.default.colors.red2,
      onClick: onClick
    },
    validate: [validateEmail]
  });
}).addWithJSX('Footer Subscription with background color option', function () {
  return _react.default.createElement(_InputFooterSubscription.default, {
    name: "input",
    disabled: (0, _addonKnobs.boolean)('Disabled', false),
    placeholder: (0, _addonKnobs.text)('Placeholder', 'Email Address (Optional)'),
    backgroundColor: _theme.default.colors.red3,
    button: {
      label: 'Enter',
      backgroundColor: _theme.default.colors.red2,
      onClick: onClick
    },
    validate: [validateEmail]
  });
});
