// mmn/atoms/LazyImage.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _jquery = _interopRequireDefault(require("jquery"));

var _uuid = require("uuid");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n    display: unset;\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  display: none;\n  ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n    ", "\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  opacity: 0.8;\n  background-color: ", ";\n  background-size: cover;\n  background-repeat: no-repeat;\n  transition: opacity 0.5s;\n  background-position: center center;\n  object-fit: cover;\n  width: 100%;\n  margin: 0 0;\n  height: 0;\n  ", ";\n  padding-bottom: ", "%;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  width: 100%;\n  max-width: ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var Wrapper = _styledComponents.default.div(_templateObject(), function (p) {
  return p.theme.breakSizes.max;
});

var Img = _styledComponents.default.img(_templateObject2(), function (p) {
  return p.theme.colors.black;
}, function (p) {
  return p.theme.breakpoint.down('s')(_templateObject3(), p.mobileSrc ? 'display: none' : 'display: unset');
}, function (p) {
  return p.height / p.width * 100;
});

var ImgMobile = (0, _styledComponents.default)(Img)(_templateObject4(), function (p) {
  return p.theme.breakpoint.down('s')(_templateObject5());
});

var LazyImage = function LazyImage(_ref) {
  var src = _ref.src,
      mobileSrc = _ref.mobileSrc,
      width = _ref.width,
      mobileWidth = _ref.mobileWidth,
      height = _ref.height,
      mobileHeight = _ref.mobileHeight,
      id = _ref.id,
      rest = _objectWithoutPropertiesLoose(_ref, ["src", "mobileSrc", "width", "mobileWidth", "height", "mobileHeight", "id"]);

  var imageId = id || (0, _uuid.v4)();
  (0, _react.useEffect)(function () {
    var img = new Image();
    var imgMobile = new Image();

    img.onload = function () {
      (0, _jquery.default)("#" + imageId).attr('src', src);
      (0, _jquery.default)("#" + imageId).css({
        paddingBottom: 0,
        opacity: 1,
        height: 'auto',
        maxWidth: '100%',
        maxHeight: '100%'
      });
    };

    img.src = src;

    imgMobile.onload = function () {
      (0, _jquery.default)("#" + imageId + "-mobile").attr('src', mobileSrc);
      (0, _jquery.default)("#" + imageId + "-mobile").css({
        paddingBottom: 0,
        opacity: 1,
        height: 'auto'
      });
    };

    img.src = src;
    imgMobile.src = mobileSrc;
    return function () {};
  }, [src, mobileSrc]);
  return _react.default.createElement(Wrapper, null, _react.default.createElement(Img, _extends({
    mobileSrc: mobileSrc,
    id: imageId,
    width: width,
    height: height
  }, rest)), mobileSrc && _react.default.createElement(ImgMobile, _extends({
    id: imageId + "-mobile",
    width: mobileWidth || width,
    height: mobileHeight || height
  }, rest)));
};

LazyImage.propTypes = {
  src: _propTypes.default.string.isRequired,
  width: _propTypes.default.number.isRequired,
  height: _propTypes.default.number.isRequired,
  mobileSrc: _propTypes.default.string,
  mobileWidth: _propTypes.default.string,
  mobileHeight: _propTypes.default.string,
  id: _propTypes.default.string
};
LazyImage.defaultProps = {
  mobileSrc: '',
  mobileWidth: '',
  mobileHeight: '',
  id: ''
};
var _default = LazyImage;
exports.default = _default;
