// mmn/atoms/Options.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var _reduxForm = require("redux-form");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _reactRedux = require("react-redux");

var _perfectScrollbar = _interopRequireDefault(require("perfect-scrollbar"));

require("../assets/css/perfect-scrollbar.css");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: 40px;\n  width: 100%;\n  height: auto;\n  max-height: 200px;\n  overflow-y: auto;\n  border: 1px solid #eee;\n  border-radius: 5px;\n  box-sizing: border-box;\n\n  div {\n    height: 40px;\n    padding-left: 10px;\n    border-bottom: 1px dotted #ddd;\n    background-color: ", ";\n  }\n\n  div:last-child {\n    border: 0;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var Container = _styledComponents.default.div(_templateObject(), function (props) {
  return props.theme.colors.gray2;
});

var Options =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(Options, _React$Component);

  function Options() {
    return _React$Component.apply(this, arguments) || this;
  }

  var _proto = Options.prototype;

  _proto.componentDidMount = function componentDidMount() {
    var id = this.props.id; // eslint-disable-next-line

    new _perfectScrollbar.default(document.querySelector("#scroll" + id));
  };

  _proto.render = function render() {
    var _this$props = this.props,
        id = _this$props.id,
        options = _this$props.options,
        changeAction = _this$props.changeAction,
        formName = _this$props.formName,
        name = _this$props.name,
        onClose = _this$props.onClose;
    return _react.default.createElement(Container, {
      id: "scroll" + id
    }, options.map(function (x) {
      return _react.default.createElement("div", {
        onKeyDown: function onKeyDown() {},
        onClick: function onClick() {
          changeAction(name, x.value, formName);
          onClose();
        },
        role: "button",
        tabIndex: "0",
        key: id + "-" + x.value
      }, x.label);
    }));
  };

  return Options;
}(_react.default.Component);

_defineProperty(Options, "propTypes", {
  id: _propTypes.PropTypes.string.isRequired,
  name: _propTypes.PropTypes.string.isRequired,
  formName: _propTypes.PropTypes.string.isRequired,
  changeAction: _propTypes.PropTypes.func.isRequired,
  onClose: _propTypes.PropTypes.func.isRequired,
  options: _propTypes.PropTypes.arrayOf(_propTypes.PropTypes.shape({
    value: _propTypes.PropTypes.string,
    label: _propTypes.PropTypes.string
  })).isRequired
});

var mdtp = function mdtp(dispatch) {
  return {
    changeAction: function changeAction(field, value, formName) {
      return dispatch((0, _reduxForm.change)(formName, field, value));
    }
  };
};

var _default = (0, _reactRedux.connect)(null, mdtp)(Options);

exports.default = _default;
