// mmn/atoms/Select.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var _reduxForm = require("redux-form");

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _uuid = _interopRequireDefault(require("uuid"));

var _jquery = _interopRequireDefault(require("jquery"));

var _reactRedux = require("react-redux");

var _Options = _interopRequireDefault(require("./Options"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n  font-style: italic;\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: 50%;\n  right: 10px;\n  width: 9px;\n  height: 6px;\n  transform: translateY(-50%);\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n      background-color: #efefef;\n      color: ", ";\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  position: relative;\n  width: 100%;\n  height: 100%;\n  background-color: ", ";\n  border: 1px solid #eee;\n  border-radius: 5px;\n  padding-left: 10px;\n  box-sizing: border-box;\n  transition: all 0.2s;\n  ", "\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  background-color: #fff;\n  font-size: 11px;\n  font-family: 'Avenir Next Medium', sans-serif !important;\n  line-height: 30px;\n  top: 5px;\n  left: 5px;\n  width: calc(100% - 40px);\n  border: 0;\n  display: ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  display: none;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  position: relative;\n  width: 100%;\n  height: 40px;\n  font-size: 11px;\n  font-family: 'Avenir Next Medium', sans-serif !important;\n  cursor: pointer;\n  line-height: 40px;\n  color: ", ";\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var ArrowSrc = function ArrowSrc(props) {
  return _react.default.createElement("svg", props, _react.default.createElement("path", {
    d: "M4.278 6l4.278-6H0l4.278 6z",
    fill: "#585858"
  }));
};

ArrowSrc.defaultProps = {
  width: "9",
  height: "6",
  viewBox: "0 0 9 6",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
};

var Container = _styledComponents.default.div(_templateObject(), function (props) {
  return props.theme.colors.gray6;
}, function (props) {
  return props.style && (0, _styledComponents.css)(_templateObject2(), props.style);
});

var Hidden = _styledComponents.default.input(_templateObject3());

var Visible = _styledComponents.default.input(_templateObject4(), function (props) {
  return props.open ? 'visible' : 'none';
});

var Input = _styledComponents.default.div(_templateObject5(), function (props) {
  return props.backgroundColor;
}, function (props) {
  return props.disabled && (0, _styledComponents.css)(_templateObject6(), props.theme.colors.gray6);
});

var Arrow = _styledComponents.default.img(_templateObject7());

var Placeholder = _styledComponents.default.span(_templateObject8());

var Select =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(Select, _React$Component);

  function Select(props) {
    var _this;

    _this = _React$Component.call(this, props) || this;

    _defineProperty(_assertThisInitialized(_this), "onMouseUp", function (e) {
      var id = _this.props.id;
      var container = (0, _jquery.default)("#" + id);
      if (!container.is(e.target) && container.has(e.target).length === 0) _this.close();
    });

    _defineProperty(_assertThisInitialized(_this), "open", function () {
      var id = _this.props.id;

      _this.setState({
        open: true
      }, function () {
        (0, _jquery.default)("#placeholder" + id).focus();
      });
    });

    _defineProperty(_assertThisInitialized(_this), "close", function () {
      _this.setState({
        open: false
      });
    });

    _this.state = {
      open: false,
      query: ''
    };
    return _this;
  }

  var _proto = Select.prototype;

  _proto.componentDidMount = function componentDidMount() {
    (0, _jquery.default)(document).on('mouseup', this.onMouseUp);
  };

  _proto.componentWillUnmount = function componentWillUnmount() {
    (0, _jquery.default)(document).off('mouseup', this.onMouseUp);
  };

  _proto.render = function render() {
    var _this2 = this;

    var _this$props = this.props,
        id = _this$props.id,
        style = _this$props.style,
        selected = _this$props.selected,
        disabled = _this$props.disabled,
        formName = _this$props.formName,
        name = _this$props.input.name,
        backgroundColor = _this$props.backgroundColor,
        placeholder = _this$props.placeholder;
    var _this$state = this.state,
        open = _this$state.open,
        query = _this$state.query;
    var options = this.props.options;
    if (query.length > 0) options = options.filter(function (x) {
      return x.label.toLowerCase().includes(query.toLowerCase());
    });
    if (options.length < 1) options.push({
      value: null,
      label: 'No Result'
    });
    return _react.default.createElement(Container, {
      id: id,
      style: style
    }, _react.default.createElement(Input, {
      open: open,
      disabled: disabled,
      focus: open,
      backgroundColor: backgroundColor,
      onClick: function onClick() {
        if (disabled) return;
        if (open) _this2.close();else _this2.open();
      }
    }, selected, " ", !selected && _react.default.createElement(Placeholder, null, placeholder), _react.default.createElement(Arrow, {
      src: ArrowSrc,
      alt: "Select"
    })), _react.default.createElement(Hidden, {
      type: "text",
      name: name,
      value: selected
    }), _react.default.createElement(Visible, {
      type: "text",
      open: open,
      value: query,
      onChange: function onChange(e) {
        _this2.setState({
          query: e.target.value
        });
      },
      id: "placeholder" + id
    }), open && _react.default.createElement(_Options.default, {
      name: name,
      formName: formName,
      options: options,
      onClose: function onClose() {
        _this2.close();
      }
    }));
  };

  return Select;
}(_react.default.Component);

_defineProperty(Select, "propTypes", {
  id: _propTypes.PropTypes.string,
  disabled: _propTypes.PropTypes.bool,
  placeholder: _propTypes.PropTypes.string,
  style: _propTypes.PropTypes.shape(_propTypes.PropTypes.any),
  input: _propTypes.PropTypes.shape(_propTypes.PropTypes.shape({
    name: _propTypes.PropTypes.string
  })).isRequired,
  selected: _propTypes.PropTypes.string,
  formName: _propTypes.PropTypes.string.isRequired,
  backgroundColor: _propTypes.PropTypes.string,
  options: _propTypes.PropTypes.arrayOf(_propTypes.PropTypes.shape({
    value: _propTypes.PropTypes.string,
    label: _propTypes.PropTypes.string
  })).isRequired
});

_defineProperty(Select, "defaultProps", {
  id: (0, _uuid.default)(),
  disabled: false,
  placeholder: null,
  style: null,
  backgroundColor: '#fafafa',
  selected: null
});

var mstp = function mstp(state, ownProps) {
  return {
    selected: (0, _reduxForm.formValueSelector)(ownProps.formName)(state, ownProps.name)
  };
};

var _default = (0, _reactRedux.connect)(mstp, null)(function (props) {
  return _react.default.createElement(_reduxForm.Field, _extends({}, props, {
    component: Select
  }));
});

exports.default = _default;
