// mmn/atoms/sidebar/SidebarHeaderTab.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: 0;\n  right: -28px;\n  width: 20px;\n  height: 20px;\n  background-color: ", ";\n  font-size: 11px;\n  color: #fff;\n  text-align: center;\n  line-height: 20px;\n  border-radius: 50%;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  position: relative;\n  height: 25px;\n  font-size: 16px;\n  font-weight: 500;\n  border-bottom: ", ";\n  color: #000;\n  display: inline-block;\n  margin-right: ", ";\n  cursor: pointer;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var Content = _styledComponents.default.div(_templateObject(), function (props) {
  return props.active ? '3px solid #A20000' : '';
}, function (props) {
  return props.quantity ? '44px' : '20px';
});

var Quantity = _styledComponents.default.div(_templateObject2(), function (props) {
  return props.theme.colors.red2;
});

var SidebarHeaderTab = function SidebarHeaderTab(_ref) {
  var label = _ref.label,
      page = _ref.page,
      activePage = _ref.activePage,
      action = _ref.action,
      quantity = _ref.quantity;
  return _react.default.createElement(Content, {
    active: page === activePage,
    quantity: quantity,
    onClick: function onClick() {
      if (action) action();
    }
  }, label, quantity && _react.default.createElement(Quantity, null, quantity));
};

SidebarHeaderTab.propTypes = {
  label: _propTypes.default.string.isRequired,
  page: _propTypes.default.string.isRequired,
  activePage: _propTypes.default.string.isRequired,
  action: _propTypes.default.func.isRequired,
  quantity: _propTypes.default.string
};
SidebarHeaderTab.defaultProps = {
  quantity: null
};
var _default = SidebarHeaderTab;
exports.default = _default;
