// @intellihr/styled-components-breakpoint/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getMedia = exports.getMediaShorthands = exports.getMixin = exports.mediaRules = exports.getNextMedia = exports.getSmallestMedia = exports.mediaTemplate = exports.ruleTemplate = exports.mediaWidthRule = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _templateObject = _taggedTemplateLiteral(['\n      ', '{\n        ', '\n      }\n    '], ['\n      ', '{\n        ', '\n      }\n    ']);

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

var _styledComponents = require('styled-components');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

// Breakpoint


// Flow-types
var mediaWidthRule = exports.mediaWidthRule = function mediaWidthRule(rule) {
  return ({ up: 'min', down: 'max' }[rule] || 'min') + '-width';
};

var ruleTemplate = exports.ruleTemplate = function ruleTemplate(rule, width) {
  return '(' + rule + ': ' + width + 'px)';
};
var mediaTemplate = exports.mediaTemplate = function mediaTemplate(rules) {
  return '@media only screen and ' + rules;
};

var getSmallestMedia = exports.getSmallestMedia = function getSmallestMedia(breakpoints) {
  var keys = Object.keys(breakpoints);

  return Number(keys.reduce(function (acc, key) {
    if (acc) {
      return acc > breakpoints[key] ? breakpoints[key] : acc;
    }
    return breakpoints[key];
  }, undefined));
};

var getNextMedia = exports.getNextMedia = function getNextMedia(breakpoints, width) {
  var keys = Object.keys(breakpoints);

  var nextBreakpoint = Number(keys.reduce(function (acc, key) {
    var current = breakpoints[key];
    if (acc && width - current <= 0) {
      return width - acc > Math.abs(width - current) || width - acc === 0 ? current : acc;
    }
    return getSmallestMedia(breakpoints);
  }, undefined));

  if (nextBreakpoint === width) {
    throw new Error('The breakpoint of value ' + nextBreakpoint + ' is the heighst, did you mean to use "[media].up"?');
  }
  return nextBreakpoint;
};

var getBound = function getBound(breakpoints, boundKey) {
  var upperBound = (0, _lodash2.default)(breakpoints).values().sortBy().find(function (breakpoint) {
    return breakpoint > breakpoints[boundKey];
  });

  if (!upperBound) {
    throw new Error('The breakpoint of value ' + boundKey + ' is the heighst, did you mean to use "[media].up"?');
  }

  return upperBound ? upperBound - 1 : false;
};

var getWidthKeyAndBoundKey = function getWidthKeyAndBoundKey(breakpoints, widthKey, boundKey) {
  var keys = {
    widthKey: widthKey,
    boundKey: boundKey
  };

  /**
   * Swap keys if width > bound
   */
  if (boundKey) {
    var width = breakpoints[widthKey];
    var bound = breakpoints[boundKey];

    if (width > bound) {
      keys = {
        widthKey: boundKey,
        boundKey: widthKey
      };
    }
  }

  return keys;
};

var mediaRules = exports.mediaRules = function mediaRules(breakpoints, initWidthKey, rule, initBoundKey) {
  var _getWidthKeyAndBoundK = getWidthKeyAndBoundKey(breakpoints, initWidthKey, initBoundKey),
      widthKey = _getWidthKeyAndBoundK.widthKey,
      boundKey = _getWidthKeyAndBoundK.boundKey;

  var width = breakpoints[widthKey];
  if (rule === 'down') {
    try {
      width = getBound(breakpoints, widthKey);
    } catch (e) {
      throw new Error('The breakpoint of value ' + widthKey + ' is the heighst, do you need breakpoints?');
    }
  }

  var bound = boundKey ? getBound(breakpoints, boundKey) : false;

  var baseWidthRule = mediaWidthRule(rule);
  var boudWidthRule = void 0;

  var baseRule = ruleTemplate(baseWidthRule, width);
  var boundRule = void 0;

  if (bound && width) {
    // Get correct rule based on width relative to bound
    baseWidthRule = mediaWidthRule('up');
    boudWidthRule = mediaWidthRule('down');

    baseRule = ruleTemplate(baseWidthRule, width);
    boundRule = ruleTemplate(boudWidthRule, bound);
    return [].concat([baseRule], bound ? [boundRule] : []).join(' and ');
  }

  if (!bound && rule === 'only') {
    // Get correct rule based on width relative to bound
    boudWidthRule = mediaWidthRule('down');
    boundRule = ruleTemplate(boudWidthRule, getNextMedia(breakpoints, width) - 1);
  }

  return [].concat([baseRule], boundRule ? [boundRule] : []).join(' and ');
};

var getMixin = exports.getMixin = function getMixin(breakpoints, key) {
  var rule = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 'up';
  return function (bound) {
    return function () {
      return (0, _styledComponents.css)(_templateObject, mediaTemplate(mediaRules(breakpoints, key, rule, bound)), _styledComponents.css.apply(undefined, arguments));
    };
  };
};

var getMediaShorthands = exports.getMediaShorthands = function getMediaShorthands(breakpoints, rule) {
  return Object.keys(breakpoints).reduce(function (acc, key) {
    return _extends({}, acc, _defineProperty({}, key, getMixin(breakpoints, key, rule)()));
  }, {});
};

var getMedia = exports.getMedia = function getMedia(breakpoints) {
  var mediasUp = getMediaShorthands(breakpoints, 'up');
  var list = Object.keys(breakpoints);

  return _extends({}, mediasUp, {
    up: function up(widthKey) {
      return getMixin(breakpoints, widthKey, 'up')();
    },
    down: function down(widthKey) {
      return getMixin(breakpoints, widthKey, 'down')();
    },
    only: function only(widthKey, boundKey) {
      return getMixin(breakpoints, widthKey, 'only')(boundKey);
    },
    list: list
  });
};

exports.default = getMedia;
