// mmn/assets/components/Minus.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _theme = _interopRequireDefault(require("../../theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Minus = function Minus(_ref) {
  var color = _ref.color,
      size = _ref.size,
      onClick = _ref.onClick;
  return _react.default.createElement("svg", {
    width: size * 10,
    height: size * 2,
    viewBox: "0 0 10 2",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg",
    onClick: onClick
  }, _react.default.createElement("rect", {
    width: size * 10,
    height: size * 2,
    rx: "1",
    fill: _theme.default.colors[color] || color
  }));
};

Minus.propTypes = {
  color: _propTypes.default.string,
  size: _propTypes.default.number,
  onClick: _propTypes.default.func
};
Minus.defaultProps = {
  color: 'black',
  size: 1,
  onClick: function onClick() {}
};
var _default = Minus;
exports.default = _default;
