// mmn/assets/components/Plus.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _theme = _interopRequireDefault(require("../../theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Plus = function Plus(_ref) {
  var color = _ref.color,
      size = _ref.size,
      onClick = _ref.onClick;
  return _react.default.createElement("svg", {
    width: size,
    height: size,
    viewBox: "0 0 11 11",
    version: "1.1",
    xmlns: "http://www.w3.org/2000/svg",
    onClick: onClick
  }, _react.default.createElement("g", {
    id: "Symbols",
    stroke: "none",
    strokeWidth: "1",
    fill: "none",
    fillRule: "evenodd"
  }, _react.default.createElement("path", {
    d: "M5.1875,5.8125 L10.3125,5.8125 C10.139911,5.8125 10,5.67258898 10,5.5 C10,5.32741102 10.139911,5.1875 10.3125,5.1875 L0.6875,5.1875 C0.860088984,5.1875 1,5.32741102 1,5.5 C1,5.67258898 0.860088984,5.8125 0.6875,5.8125 L5.1875,5.8125 L5.1875,10.3125 C5.1875,10.139911 5.32741102,10 5.5,10 C5.67258898,10 5.8125,10.139911 5.8125,10.3125 L5.8125,0.6875 C5.8125,0.860088984 5.67258898,1 5.5,1 C5.32741102,1 5.1875,0.860088984 5.1875,0.6875 L5.1875,5.8125 Z",
    id: "Combined-Shape",
    stroke: _theme.default.colors[color] || color,
    strokeWidth: "2",
    fill: _theme.default.colors[color] || color,
    fillRule: "nonzero"
  })));
};

Plus.propTypes = {
  color: _propTypes.default.string,
  size: _propTypes.default.number,
  onClick: _propTypes.default.func
};
Plus.defaultProps = {
  color: 'black',
  size: 9,
  onClick: function onClick() {}
};
var _default = Plus;
exports.default = _default;
