// @contentful/rich-text-types/dist/rich-text-types.es5.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

/**
 * Map of all Contentful block types. Blocks contain inline or block nodes.
 */
var BLOCKS = {
    DOCUMENT: 'document',
    PARAGRAPH: 'paragraph',
    HEADING_1: 'heading-1',
    HEADING_2: 'heading-2',
    HEADING_3: 'heading-3',
    HEADING_4: 'heading-4',
    HEADING_5: 'heading-5',
    HEADING_6: 'heading-6',
    OL_LIST: 'ordered-list',
    UL_LIST: 'unordered-list',
    LIST_ITEM: 'list-item',
    HR: 'hr',
    QUOTE: 'blockquote',
    EMBEDDED_ENTRY: 'embedded-entry-block',
    EMBEDDED_ASSET: 'embedded-asset-block',
};

/**
 * Map of all Contentful inline types. Inline contain inline or text nodes.
 */
var inlines = {
    HYPERLINK: 'hyperlink',
    ENTRY_HYPERLINK: 'entry-hyperlink',
    ASSET_HYPERLINK: 'asset-hyperlink',
    EMBEDDED_ENTRY: 'embedded-entry-inline',
};

/**
 * Map of all Contentful marks.
 */
var marks = {
    BOLD: 'bold',
    ITALIC: 'italic',
    UNDERLINE: 'underline',
    CODE: 'code',
};

var _a;
/**
 * Array of all top level block types.
 * Only these block types can be the direct children of the document.
 */
var TOP_LEVEL_BLOCKS = [
    BLOCKS.PARAGRAPH,
    BLOCKS.HEADING_1,
    BLOCKS.HEADING_2,
    BLOCKS.HEADING_3,
    BLOCKS.HEADING_4,
    BLOCKS.HEADING_5,
    BLOCKS.HEADING_6,
    BLOCKS.OL_LIST,
    BLOCKS.UL_LIST,
    BLOCKS.HR,
    BLOCKS.QUOTE,
    BLOCKS.EMBEDDED_ENTRY,
    BLOCKS.EMBEDDED_ASSET,
];
/**
 * Array of all void block types
 */
var VOID_BLOCKS = [BLOCKS.HR, BLOCKS.EMBEDDED_ENTRY, BLOCKS.EMBEDDED_ASSET];
/**
 * Dictionary of all container block types, and the set block types they accept as children.
 */
var CONTAINERS = (_a = {},
    _a[BLOCKS.OL_LIST] = [BLOCKS.LIST_ITEM],
    _a[BLOCKS.UL_LIST] = [BLOCKS.LIST_ITEM],
    _a[BLOCKS.LIST_ITEM] = TOP_LEVEL_BLOCKS.slice(),
    _a[BLOCKS.QUOTE] = [BLOCKS.PARAGRAPH],
    _a);

exports.BLOCKS = BLOCKS;
exports.INLINES = inlines;
exports.MARKS = marks;
exports.TOP_LEVEL_BLOCKS = TOP_LEVEL_BLOCKS;
exports.VOID_BLOCKS = VOID_BLOCKS;
exports.CONTAINERS = CONTAINERS;
