// mmn/atoms/input/InputCommon.js
"use strict";

exports.__esModule = true;
exports.Action = exports.Error = exports.Button = exports.Icon = exports.HoverContainer = exports.Container = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: 0;\n  right: 10px;\n\n  span {\n    height: 100%;\n    line-height: 50px;\n    padding-right: 5px;\n    color: ", ";\n    font-size: 11px;\n    cursor: pointer;\n  }\n\n  div {\n    position: absolute;\n    top: 15px;\n    right: 0;\n    width: 20px;\n    height: 20px;\n\n    img {\n      width: 100%;\n      height: 100%;\n      object-fit: contain;\n      cursor: pointer;\n    }\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n  margin-top: 5px;\n  color: ", ";\n  font-size: 11px;\n  margin-left: 3px;\n\n  img {\n    margin-right: 5px;\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: 0;\n  right: 0;\n  width: 105px;\n  height: 50px;\n  text-align: center;\n  border-top-right-radius: 5px;\n  border-bottom-right-radius: 5px;\n  line-height: 40px;\n  font-size: 14px;\n  cursor: pointer;\n  border: 0;\n\n  &:disabled {\n    cursor: not-allowed;\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: 15px;\n  left: 11px;\n  width: 20px;\n  height: 20px;\n\n  img {\n    width: 100%;\n    height: 100%;\n    object-fit: contain;\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  position: relative;\n  width: 100%;\n  min-height: 50px;\n  ", "\n\n  input {\n    width: 100%;\n    height: 50px;\n    background-color: ", ";\n    border: 1px solid #d9d9d6;\n    border-radius: 5px;\n    font-family: 'Avenir Next', sans-serif !important;\n    font-size: 12px;\n    color: ", ";\n    padding-left: ", ";\n    box-sizing: border-box;\n    transition: all 0.5s;\n\n    &:focus {\n      border-color: ", ";\n      padding-top: 15px;\n\n      ::placeholder {\n        color: transparent;\n        transition: all 0.5s;\n      }\n    }\n\n    &:disabled {\n      background-color: #efefef;\n      color: ", ";\n    }\n  }\n\n  label {\n    position: absolute;\n    top: 7px;\n    left: ", ";\n    font-size: 11px;\n    color: ", ";\n    transition: all 0.5s;\n    opacity: 0;\n    pointer-events: none;\n  }\n\n  .inputFocus {\n    color: ", ";\n    padding-top: 15px;\n  }\n\n  .labelFocus {\n    color: ", ";\n    opacity: 1;\n  }\n\n  .inputBlur {\n    color: ", ";\n  }\n\n  .labelBlur {\n    color: ", ";\n  }\n\n  .noIcon {\n    left: 0;\n    padding-left: 10px;\n  }\n\n  ", "\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n      ", "\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  position: relative;\n  width: 100%;\n  min-height: 50px;\n  ", "\n\n  input {\n    width: 100%;\n    height: 50px;\n    background-color: ", ";\n    border: 1px solid #d9d9d6;\n    border-radius: 5px;\n    font-family: 'Avenir Next', sans-serif !important;\n    font-size: 12px;\n    color: ", ";\n    padding-left: ", ";\n    box-sizing: border-box;\n    transition: all 0.2s;\n\n    &:disabled {\n      background-color: #efefef;\n      color: ", ";\n    }\n\n    &:focus {\n      border-color: ", ";\n      padding-top: 15px;\n\n      ::placeholder {\n        color: transparent;\n        transition: all 0.5s;\n      }\n    }\n  }\n\n  label {\n    position: absolute;\n    top: 7px;\n    left: ", ";\n    font-size: 11px;\n    color: ", ";\n    transition: all 0.5s;\n    opacity: 0;\n    pointer-events: none;\n  }\n\n  .inputFocus {\n    color: ", ";\n    padding-top: 15px;\n  }\n\n  .labelFocus {\n    color: ", ";\n    opacity: 1;\n  }\n\n  .inputBlur {\n    color: ", ";\n  }\n\n  .labelBlur {\n    color: ", ";\n  }\n\n  .noIcon {\n    left: 0;\n    padding-left: 10px;\n  }  \n\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var Container = _styledComponents.default.div(_templateObject(), function (props) {
  return props.style && (0, _styledComponents.css)(_templateObject2(), props.style);
}, function (props) {
  return props.backgroundColor ? props.backgroundColor : '#fafafa';
}, function (props) {
  return props.theme.colors.gray45;
}, function (props) {
  return props.icon ? '40px' : '10px';
}, function (props) {
  return props.theme.colors.gray45;
}, function (props) {
  return props.theme.colors.red2;
}, function (p) {
  return p.icon ? '41px' : '11px';
}, function (props) {
  return props.theme.colors.red2;
}, function (props) {
  return props.theme.colors.gray45;
}, function (props) {
  return props.theme.colors.red2;
}, function (props) {
  return props.theme.colors.gray6;
}, function (props) {
  return props.theme.colors.gray45;
}, function (p) {
  return p.theme.autofill();
});

exports.Container = Container;

var HoverContainer = _styledComponents.default.div(_templateObject3(), function (props) {
  return props.style && (0, _styledComponents.css)(_templateObject4(), props.style);
}, function (props) {
  return props.backgroundColor ? props.backgroundColor : '#fafafa';
}, function (props) {
  return props.theme.colors.gray45;
}, function (props) {
  return props.icon ? '40px' : '10px';
}, function (props) {
  return props.theme.colors.red2;
}, function (props) {
  return props.theme.colors.gray45;
}, function (p) {
  return p.icon ? '41px' : '11px';
}, function (props) {
  return props.theme.colors.red2;
}, function (props) {
  return props.theme.colors.gray45;
}, function (props) {
  return props.theme.colors.red2;
}, function (props) {
  return props.theme.colors.gray6;
}, function (props) {
  return props.theme.colors.gray45;
}, function (p) {
  return p.theme.autofill();
});

exports.HoverContainer = HoverContainer;

var Icon = _styledComponents.default.div(_templateObject5());

exports.Icon = Icon;

var Button = _styledComponents.default.button(_templateObject6());

exports.Button = Button;

var Error = _styledComponents.default.div(_templateObject7(), function (props) {
  return props.theme.colors.red2;
});

exports.Error = Error;

var Action = _styledComponents.default.div(_templateObject8(), function (props) {
  return props.color ? props.color : props.theme.colors.gray45;
});

exports.Action = Action;
