// mmn/molecules/ItemCarousel.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = _interopRequireDefault(require("../theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  width: 100%;\n  justify-content: center;\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n  background: ", ";\n  width: 10px;\n  height: 10px;\n  border-radius: 50%;\n  display: ", ";\n  margin: 5px;\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  justify-content: center;\n  align-items: center;\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose([""]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n  :hover {\n    cursor: pointer;\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  width: 300px;\n  height: 150px;\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  background: #b9b9b9;\n  border-radius: 5px;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  background: ", ";\n  padding: 13px 27px;\n  border-radius: 5px;\n  margin: 0 10px;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  font-size: 14px;\n  line-height: 19px;\n  text-align: center;\n  max-width: 665px;\n  color: ", ";\n  margin-bottom: 30px;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  height: 38px;\n  font-size: 28px;\n  line-height: 38px;\n  display: flex;\n  align-items: center;\n  text-align: center;\n  color: ", ";\n  margin-bottom: 9px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  align-items: center;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/* eslint-disable */
var CircleArrowIcon = function CircleArrowIcon(_ref) {
  var direction = _ref.direction,
      size = _ref.size,
      color = _ref.color,
      secondaryColor = _ref.secondaryColor;
  var directionDict = {
    right: 0,
    down: 90,
    left: 180,
    up: 271
  };
  return _react.default.createElement("svg", {
    width: 7 * +size,
    height: 12 * +size,
    viewBox: "0 0 7 12",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, _react.default.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M6.70719 11.707C7.09766 11.3164 7.09755 10.6832 6.70697 10.2928L2.41551 6.00272L6.70711 1.71112C7.09763 1.3206 7.09763 0.687434 6.70711 0.296909C6.31658 -0.0936146 5.68342 -0.0936156 5.29289 0.296909L0.296924 5.29288C0.144669 5.44513 0.0517736 5.63427 0.0182391 5.83153C-0.0360565 6.14169 0.0566034 6.47254 0.296201 6.71206L5.29298 11.7072C5.68357 12.0977 6.31673 12.0976 6.70719 11.707Z",
    fill: _theme.default[color] || color
  }));
};

CircleArrowIcon.propTypes = {
  direction: _propTypes.default.string,
  size: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number]),
  color: _propTypes.default.string,
  secondaryColor: _propTypes.default.string
};
CircleArrowIcon.defaultProps = {
  direction: 'left',
  size: '25.5',
  color: 'red2',
  secondaryColor: 'white'
};

var CircleArrowButton = function CircleArrowButton(_ref2) {
  var type = _ref2.type;
  return type === 'left';
};

var AddCarouselState =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(AddCarouselState, _Component);

  function AddCarouselState() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "state", {
      current: 0
    });

    _defineProperty(_assertThisInitialized(_this), "moveLeft", function (amount) {
      var moveBy = _this.props.moveBy;

      if (!moveBy) {
        if (typeof amount !== 'number') moveBy = 1;
      }

      var current = _this.state.current;
      if (current - moveBy >= 0) _this.setState({
        current: current - moveBy
      });
    });

    _defineProperty(_assertThisInitialized(_this), "moveRight", function (amount) {
      var moveBy = _this.props.moveBy;

      if (!moveBy) {
        if (typeof amount !== 'number') moveBy = 1;
      }

      var total = _this.props.total;
      var current = _this.state.current;
      if (current + moveBy < total) _this.setState({
        current: current + moveBy
      });
    });

    _defineProperty(_assertThisInitialized(_this), "setCurrent", function (next) {
      return _this.setState({
        current: next
      });
    });

    return _this;
  }

  var _proto = AddCarouselState.prototype;

  _proto.render = function render() {
    var current = this.state.current;
    var moveLeft = this.moveLeft,
        moveRight = this.moveRight,
        setCurrent = this.setCurrent;
    var children = this.props.children;
    return children({
      moveLeft: moveLeft,
      moveRight: moveRight,
      current: current,
      setCurrent: setCurrent
    });
  };

  return AddCarouselState;
}(_react.Component);

var Header = _styledComponents.default.div(_templateObject());

var Heading = _styledComponents.default.div(_templateObject2(), function (p) {
  return p.theme.colors.black;
});

var Subheading = _styledComponents.default.div(_templateObject3(), function (p) {
  return p.theme.colors.gray45;
});

var ItemWrapper = _styledComponents.default.div(_templateObject4(), function (p) {
  return p.theme.colors.white;
});

var Square = _styledComponents.default.div(_templateObject5());

var TempArrowWrapper = _styledComponents.default.div(_templateObject6());

var TempArrow = function TempArrow(_ref3) {
  var direction = _ref3.direction,
      onClick = _ref3.onClick;
  return direction === 'left' ? _react.default.createElement(TempArrowWrapper, {
    onClick: onClick
  }, "<--") : _react.default.createElement(TempArrowWrapper, {
    onClick: onClick
  }, "-->");
};

var Item = function Item(_ref4) {
  var content = _ref4.content;
  return _react.default.createElement(ItemWrapper, null, _react.default.createElement(Square, null, content));
};

var Carousel = _styledComponents.default.div(_templateObject7());

var CarouselWrapper = _styledComponents.default.div(_templateObject8());

var TabIndex = _styledComponents.default.div(_templateObject9(), function (p) {
  return p.active ? p.theme.colors.red2 : p.theme.colors.gray65;
}, function (p) {
  return p.visible ? 'auto' : 'none';
});

var TabIndexWrapper = _styledComponents.default.div(_templateObject10());

var total = 12;
var width = 4;
var list = new Array(total).fill(Item).map(function (X, i) {
  return _react.default.createElement(X, {
    content: i
  });
});

var ShopByBrandsCarousel =
/*#__PURE__*/
function (_Component2) {
  _inheritsLoose(ShopByBrandsCarousel, _Component2);

  function ShopByBrandsCarousel() {
    return _Component2.apply(this, arguments) || this;
  }

  var _proto2 = ShopByBrandsCarousel.prototype;

  _proto2.render = function render() {
    return _react.default.createElement("div", null, _react.default.createElement(Header, null, _react.default.createElement(Heading, null, "Shop By Brand"), _react.default.createElement(Subheading, null, "MedMen distribute tons of brands and our own brands like [statemade], LuxLyte and MedMen. Enjoy your cannabis shopping with us.")), _react.default.createElement(Carousel, null, _react.default.createElement(AddCarouselState, {
      total: total,
      moveBy: width
    }, function (_ref5) {
      var moveLeft = _ref5.moveLeft,
          moveRight = _ref5.moveRight,
          current = _ref5.current,
          setCurrent = _ref5.setCurrent;
      return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(CarouselWrapper, null, _react.default.createElement(TempArrow, {
        direction: "left",
        onClick: moveLeft
      }), list.filter(function (_, i) {
        return current <= i && current + width > i;
      }), _react.default.createElement(TempArrow, {
        direction: "right",
        onClick: moveRight
      })), _react.default.createElement(TabIndexWrapper, null, list.map(function (x, i) {
        return _react.default.createElement(TabIndex, {
          active: i === current,
          visible: !(i % width),
          onClick: function onClick() {
            return setCurrent(i);
          }
        });
      })));
    })));
  };

  return ShopByBrandsCarousel;
}(_react.Component);

ShopByBrandsCarousel.propTypes = {};
var _default = ShopByBrandsCarousel;
exports.default = _default;
