// mmn/atoms/button/AgeGateButton.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _ButtonTemplate = _interopRequireDefault(require("./ButtonTemplate"));

var _ChildrenWithIcon = _interopRequireDefault(require("../ChildrenWithIcon"));

var _propTypes2 = require("./propTypes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  button {\n    border: 1px solid ", " !important;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var HijackHoverBorder = _styledComponents.default.div(_templateObject(), function (props) {
  return props.theme.colors[props.borderColor];
});

var AgeGateButton = function AgeGateButton(_ref) {
  var children = _ref.children,
      onClick = _ref.onClick,
      disabled = _ref.disabled,
      loading = _ref.loading,
      Icon = _ref.icon;
  return _react.default.createElement(HijackHoverBorder, null, _react.default.createElement(_ButtonTemplate.default, {
    bgColor: "gray8",
    textColor: "white",
    width: "100px",
    onClick: onClick,
    disabled: disabled,
    loading: loading
  }, _react.default.createElement(_ChildrenWithIcon.default, {
    icon: Icon,
    side: "right"
  }, children)));
};

AgeGateButton.propTypes = _extends({}, _propTypes2.propTypes, {
  icon: _propTypes.default.element.isRequired
});
AgeGateButton.defaultProps = _propTypes2.defaultProps;
var _default = AgeGateButton;
exports.default = _default;
