// mmn/theme/index.js
"use strict";

exports.__esModule = true;
exports.default = exports.autofill = void 0;

var _styledComponents = require("styled-components");

var _breakpoint = _interopRequireWildcard(require("./breakpoint"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n    @media (max-width: ", "em) {\n      ", "\n    }\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  input:-webkit-autofill,\n  input:-webkit-autofill:hover,\n  input:-webkit-autofill:focus,\n  textarea:-webkit-autofill,\n  textarea:-webkit-autofill:hover,\n  textarea:-webkit-autofill:focus,\n  select:-webkit-autofill,\n  select:-webkit-autofill:hover,\n  select:-webkit-autofill:focus {\n    -webkit-text-fill-color: ", ";\n    box-shadow: 0 0 0 1000px ", " inset;\n    ", "\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var autofill = function autofill() {
  for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
    args[_key] = arguments[_key];
  }

  return (0, _styledComponents.css)(_templateObject(), function (p) {
    return p.theme.colors.black;
  }, function (p) {
    return p.theme.colors.white;
  }, args && args.length && _styledComponents.css.apply(void 0, args));
}; // Iterate through the sizes and create a media template


exports.autofill = autofill;
var media = Object.keys(_breakpoint.breakSizes).reduce(function (acc, label) {
  acc[label] = function () {
    return (0, _styledComponents.css)(_templateObject2(), _breakpoint.breakSizes[label] / 16, _styledComponents.css.apply(void 0, arguments));
  };

  return acc;
}, {});
var theme = {
  breakSizes: _breakpoint.breakSizes,
  media: media,
  autofill: autofill,
  breakpoint: _breakpoint.default,
  colors: {
    green2: '#00563f',
    green1: '#479f46',
    blue2: '#000a8b',
    blue1: '#00a8e1',
    purple: '#873299',
    pink: '#ea1d76',
    yellow: '#ffb819',
    red3: '#550505',
    red2: '#aa1f16',
    red1: '#bf1c1f',
    black: '#222',
    gray8: '#22222260',
    gray7: '#22222220',
    gray6: '#585858',
    gray5: '#858585',
    gray45: '#7C7C7C',
    gray4: '#aaa',
    gray31: '#C4C4C4',
    gray3: '#c7c7c7',
    gray26: '#D2D2D2',
    gray25: '#D9D9D6',
    gray2: '#f4f4f4',
    gray1: '#fafafa',
    white: '#fff'
  },
  fonts: {
    primary: 'Avenir Next, sans-serif'
  },
  elements: {
    header: '120px',
    footer: '800px'
  }
};
var _default = theme;
exports.default = _default;
