// mmn/constants/productStrainInfo.js
"use strict";

exports.__esModule = true;
exports.default = void 0;
var _default = {
  getProductInfoFromType: function getProductInfoFromType(type) {
    if (type === void 0) {
      type = '';
    }

    var infoDict = {
      indicadominant: 'A high THC strain with a terpene profile that promotes maximum relaxation and a sense of transcendence.',
      hybrid: 'A hybrid strain is created by cross-breeding two or more cannabis strains. Hybrid may refer to indica and sativa crosses, crosses of two or more sativas, two or more indicas and other variations. Due to their bred genetics, hybrids can offer a desired profile of benefits, often elevating desirable effects while minimizing less-desirable effects depending on personal preference. Hybrids are used to help manage the symptoms of various health conditions.',
      '50/50': 'A hybrid strain is created by cross-breeding two or more cannabis strains. Hybrid may refer to indica and sativa crosses, crosses of two or more sativas, two or more indicas and other variations. Due to their bred genetics, hybrids can offer a desired profile of benefits, often elevating desirable effects while minimizing less-desirable effects depending on personal preference. Hybrids are used to help manage the symptoms of various health conditions.',
      sativa: 'Sativa strains are known for eliciting more stimulating, euphoric and uplifting effects than indica strains, making them ideal for daytime use. Characteristics may vary within flowers of the same strain. This variation may be due to the different terpene profile of individual plants.',
      indica: 'Indicas are known to evoke a sedative effect, making them popular with evening users. Indicas generally contain a higher level of Myrcene - a terpene thought to produce the "couch-lock" effect, than sativa strains. It has been proposed that the effects of specific strains are dependent on the unique terpene profile of individual plants.',
      highcbd: 'Strains that contain more than 2 percent CBD are considered to be high CBD strains. Very high CBD strains may contain over 20 percent. High CBD strains with low THC are believed to provide optimal health benefits and promote body relaxation with minimal psychoactive side effects, while strains with high CBD and high THC are believed to provide a balance of health benefits and cognitive side effects.',
      sativadominant: 'A high THC strain that promotes focus and composure. Perfect for creative endeavors.',
      default: ''
    };
    var typeKey = type.replace(/\s*/g, '').toLowerCase();
    return infoDict[typeKey] || infoDict.default;
  },
  colorDict: {
    indicadominant: '#00A8E0',
    hybrid: '#94299E',
    '50/50': '#94299E',
    sativa: '#FFBA19',
    indica: '#000A8C',
    highcbd: '#035440',
    sativadominant: '#EB1C75',
    default: '#222222'
  },
  typeNameDict: {
    indicadominant: 'Indica Dominant',
    hybrid: 'Hybrid',
    '50/50': 'Hybrid',
    sativa: 'Sativa',
    indica: 'Indica',
    highcbd: 'High CBD',
    sativadominant: 'Sativa Dominant',
    default: 'Standard'
  },
  getProductStrainColors: function getProductStrainColors(type) {
    if (type === void 0) {
      type = '';
    }

    var typeKey = type.replace(/\s*/g, '').toLowerCase();
    return this.colorDict[typeKey] || this.colorDict.default;
  },
  getProductLabelFromType: function getProductLabelFromType(type) {
    if (type === void 0) {
      type = '';
    }

    var typeKey = type.replace(/\s*/g, '').toLowerCase();
    return this.typeNameDict[typeKey] || this.typeNameDict.default;
  }
};
exports.default = _default;
